### PR TITLE
Gh 3128/sdk integration

### PR DESCRIPTION
**Pull request by MouazAlzahabi**
Handles #3128 

- Integrating CustomAuth SDK
- Temporary changes to resolve the conflicts
- TK SDK will be integrated later

### CODE REVIEW

**Review by biafra23 — APPROVED** [review]
I'll approve it anyway :-)

File: Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved [hunk 1]
```diff
@@ -181,6 +217,15 @@
           "version": null
         }
       },
+      {
+        "package": "SafeWeb3",
+        "repositoryURL": "https://github.com/safe-global/SafeWeb3.swift",
+        "state": {
+          "branch": "rename-to-SafeWeb3",
```
Comment (elgatovital) [inline Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:224]: can we set branch to null here and make SafeWeb3 available by default?

File: Multisig/Logic/Web3AuthManager.swift [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+//
+
+import Foundation
+import CustomAuth
+class Web3AuthManager {
+    static let shared = Web3AuthManager()
+
+    private init() { }
+
+    //    func loginGoogle() async {
```
Comment (elgatovital) [inline Multisig/Logic/Web3AuthManager.swift]: what's with the commented out code? As an exception there should either be specific instructions what to uncomment / adjust or the whole block should not be committed at all.

File: Multisig/UI/SwiftUI/UI Library/String+Error.swift [hunk 3]
```diff
@@ -7,9 +7,9 @@
...
 
 import Foundation
-
-extension String: LocalizedError {
-    public var errorDescription: String? {
-        self
-    }
-}
+//
+//extension String: LocalizedError {
```
Comment (elgatovital) [inline Multisig/UI/SwiftUI/UI Library/String+Error.swift]: should it be removed then?

File: Shared/Utils/Data+MultisigExtension.swift [hunk 4]
```diff
@@ -8,6 +8,39 @@ import CryptoSwift
 // MARK: - Hex String to Data conversion
 public extension Data {
 
+    static func value(of nibble: UInt8) -> UInt8? {
+        guard let letter = String(bytes: [nibble], encoding: .ascii) else { return nil }
+        return UInt8(letter, radix: 16)
+    }
+
+    // TODO: Duplicate code. Remove and use init(hex: String) instead, when possible
+    init(hexWC: String) {
```
Comment (elgatovital) [inline Shared/Utils/Data+MultisigExtension.swift:18]: what's hexWC stands for?

File: Shared/Utils/Data+MultisigExtension.swift [hunk 5]
```diff
@@ -8,6 +8,39 @@ import CryptoSwift
 // MARK: - Hex String to Data conversion
 public extension Data {
 
+    static func value(of nibble: UInt8) -> UInt8? {
+        guard let letter = String(bytes: [nibble], encoding: .ascii) else { return nil }
+        return UInt8(letter, radix: 16)
+    }
+
+    // TODO: Duplicate code. Remove and use init(hex: String) instead, when possible
```
Comment (elgatovital) [inline Shared/Utils/Data+MultisigExtension.swift]: please describe the reasoning behind this duplication.

File: Shared/Utils/Data+MultisigExtension.swift (hunk 4 above)
Comment (biafra23) [reply Shared/Utils/Data+MultisigExtension.swift:18]: WalletConnect. This is the implementation that WalletConnect pulls in.

File: Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved (hunk 1 above)
Comment (MouazAlzahabi) [reply Multisig.xcodeproj/project.xcworkspace/xcshareddata/swiftpm/Package.resolved:224]: No, I need that branch specifically, however this file is auto generated

**Pull request merged**